Patch: 10_wrapper_uid.patch
Author: Tollef Fog Heen <tfheen@debian.org>
Makes sure we're called with the right UID and GID (Closes: #36010, #89564, #89848, 89818)
Index: mailman-2.1.12/src/cgi-wrapper.c
===================================================================
--- mailman-2.1.12.orig/src/cgi-wrapper.c	2009-03-02 20:41:55.000000000 +0100
+++ mailman-2.1.12/src/cgi-wrapper.c	2009-03-02 20:49:59.000000000 +0100
@@ -42,7 +42,8 @@
         char* fake_argv[3];
 
         running_as_cgi = 1;
-        check_caller(logident, parentgroup);
+        if (getgid() >= 100 && getgid() != 65534)
+                check_caller(logident, parentgroup);
 
         /* For these CGI programs, we can ignore argc and argv since they
          * don't contain anything useful.  `script' will always be the driver
Index: mailman-2.1.12/src/mail-wrapper.c
===================================================================
--- mailman-2.1.12.orig/src/mail-wrapper.c	2009-03-02 20:41:56.000000000 +0100
+++ mailman-2.1.12/src/mail-wrapper.c	2009-03-02 20:49:59.000000000 +0100
@@ -74,7 +74,8 @@
                 fatal(logident, MAIL_ILLEGAL_COMMAND,
                       "Illegal command: %s", argv[1]);
 
-        check_caller(logident, parentgroup);
+        if (getgid() >= 100 && getgid() != 65534)
+                check_caller(logident, parentgroup);
 
         /* If we got here, everything must be OK */
         status = run_script(argv[1], argc, argv, env);
